File: app/src/main/java/com/makingiants/todayhistory/today/TodayAdapter.java
package com.makingiants.todayhistory.today;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;
import butterknife.Bind;
import butterknife.ButterKnife;
import com.makingiants.today.api.repository.history.pojo.Event;
import com.makingiants.todayhistory.R;
import com.squareup.picasso.Picasso;
import java.util.List;

public class TodayAdapter extends RecyclerView.Adapter<TodayAdapter.SongViewHolder> {
  private Event[] mEvents;
  private Picasso mPicasso;

  public TodayAdapter(Picasso picasso) {
    mPicasso = picasso;
  }

  @Override
  public SongViewHolder onCreateViewHolder(ViewGroup container, int viewType) {
    LayoutInflater inflater = LayoutInflater.from(container.getContext());
    View root = inflater.inflate(R.layout.today_item, container, false);
    return new SongViewHolder(root);
  }

  @Override
  public void onBindViewHolder(SongViewHolder holder, int position) {
    Event event = mEvents[position];

    holder.titleTextView.setText(event.getTitle());
    holder.dateTextView.setText(event.getDate());

    mPicasso.load(event.getImageUrl())
        .fit()
        .into(holder.headerImage);

    //mPicasso.load(event.getImageUrl())
    //    .fit()
    //    .into(holder.headerImage, PicassoPalette.with(event.getImageUrl(), holder.headerImage)
    //        .use(PicassoPalette.Profile.VIBRANT)
    //        .intoBackground(holder.textLinearLayout)
    //        .intoTextColor(holder.dateTextView, PicassoPalette.Swatch.BODY_TEXT_COLOR)
    //        .intoTextColor(holder.titleTextView, PicassoPalette.Swatch.TITLE_TEXT_COLOR));
  }

  public void setEvents(List<Event> events) {
    mEvents = events.toArray(new Event[events.size()]);
    notifyDataSetChanged();
  }

  @Override
  public int getItemCount() {
    return mEvents == null ? 0 : mEvents.length;
  }

  public static class SongViewHolder extends RecyclerView.ViewHolder {
    @Bind(R.id.item_today_layout_text) LinearLayout textLinearLayout;
    @Bind(R.id.item_today_text_date) TextView dateTextView;
    @Bind(R.id.item_today_text_title) TextView titleTextView;
    @Bind(R.id.item_today_image) ImageView headerImage;

    SongViewHolder(View itemView) {
      super(itemView);
      ButterKnife.bind(this, itemView);
    }
  }
}

File: app/src/main/kotlin/com/makingiants/todayhistory/today/TodayAdapter.kt
package com.makingiants.todayhistory.today

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.github.florent37.picassopalette.BitmapPalette
import com.github.florent37.picassopalette.PicassoPalette
import com.makingiants.today.api.repository.history.pojo.Event
import com.makingiants.todayhistory.R
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.today_item.view.*

class TodayAdapter(private val mPicasso: Picasso) : RecyclerView.Adapter<TodayAdapter.SongViewHolder>() {
    private var mEvents: Array<Event>? = null

    override fun onCreateViewHolder(container: ViewGroup, viewType: Int): SongViewHolder {
        val inflater = LayoutInflater.from(container.context)
        val root = inflater.inflate(R.layout.today_item, container, false)
        return SongViewHolder(root)
    }

    override fun onBindViewHolder(holder: SongViewHolder, position: Int) {
        val event = mEvents!![position]

        holder.itemView.titleTextView.text = event.title
        holder.itemView.dateTextView.text = event.date

        mPicasso.load(event.imageUrl)
                .placeholder(R.color.material_grey_600)
                .fit()
                .centerInside()
                .into(holder.itemView.image,
                        PicassoPalette.with(event.imageUrl, holder.itemView.image)
                                .use(BitmapPalette.Profile.MUTED_DARK)
                                .intoBackground(holder.itemView.image))

    }

    fun setEvents(events: List<Event>) {
        mEvents = events.toTypedArray()
        notifyDataSetChanged()
    }

    override fun getItemCount(): Int = mEvents?.size ?: 0

    class SongViewHolder constructor(itemView: View) : RecyclerView.ViewHolder(itemView) {

    }
}
